src/analysis/retail/deathknight/unholy/modules/talents/CommanderOfTheDead.tsx

```diff
@@ -0,0 +1,105 @@
+import { t } from '@lingui/macro';
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/deathknight';
+import { SpellLink } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { ApplyBuffEvent, SummonEvent } from 'parser/core/Events';
+import { When } from 'parser/core/ParseResults';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import { formatPercentage } from 'common/format';
+
+class CommanderOfTheDead extends Analyzer {
+  commanderBuffs = 0;
+  petSummons = 0;
+  petSummonNames = ['Army of the Dead', 'Dark Arbiter', 'Summon Gargoyle', 'Army of the Damned'];
+  buffedPets: string[] = [];
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS.COMMANDER_OF_THE_DEAD_TALENT);
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.COMMANDER_OF_THE_DEAD_BUFF),
+      this.onBuffEvent,
+    );
+
+    this.addEventListener(Events.summon.by(SELECTED_PLAYER), this.onSummonEvent);
+  }
+
+  onBuffEvent(event: ApplyBuffEvent) {
+    let summonId = '';
+    if (event.targetInstance) {
+      summonId = event.targetID.toString() + '|' + event.targetInstance.toString(); // This is needed since the buff sometimes applies twice to the same summon
+    } else {
+      summonId = event.targetID.toString();
+    }
+    if (!this.buffedPets.includes(summonId)) {
+      this.commanderBuffs += 1;
+      this.buffedPets.push(summonId);
+    }
+  }
+
+  onSummonEvent(event: SummonEvent) {
+    console.log(event);
+    if (this.petSummonNames.includes(event.ability.name)) {
```

Comment:
This should be done on the ability guid instead of the name.  String comparison is done as a last resort

Reply:
Had a look and the ability guids associated with the summons don't match what we've got in the files for the original spells. Want me to add in the summon versions of the spells or hard-code the ids in the list?